packages/venia-concept/local-intercept.js

```diff
@@ -16,7 +16,24 @@
  * with many customizations, this function would tap those targets and add
  * or modify functionality from its dependencies.
  */
+const { Targetables } = require('@magento/pwa-buildpack');
 
-function localIntercept() {}
+function localIntercept(targets) {
+    const myTheme = Targetables.using(targets);
+
+    const button = myTheme.reactComponent(
+        '@magento/venia-ui/lib/components/Button/button.js'
+    );
+
+    const CustomButton = button.addImport(
+        "CustomButton from '@magento/venia-concept/src/CustomButton/CustomButton.js'"
+    );
+    button.replaceJSX(
+        'button',
+        `<${CustomButton} className={rootClassName} type={type} disabled={disabled} {...restProps}>{children}</${CustomButton}>`
```

Comment:
Preferably we would just overwrite the export of the `Button` component with the `CustomButton` rather than replacing the JSX which requires knowledge of the props/children structure.